### ISSUE TITLE
Change to initialValues causes user input to be lost

### ISSUE THREAD

**Issue by kpdecker** [erikras/redux-form#832]
If the initialValues object changes, non-conflicting user changes will be lost as the entire form is reset. While the correct behavior for the conflicting field case is a bit ambiguous, it seems like we should not invalidate user input in the event this changed.

```
export default reduxForm(GroupSettings.formProps,
/* istanbul ignore next */
(state) => {
  return {
    initialValues: state.group
  };
})(GroupSettings);
```

**Comment by kpdecker** [comment]
```
npm ls | grep redux-form
├─┬ redux-form@4.2.2
```

**Comment by erikras** [comment]
We had a pretty long discussion about this on #370.

**Comment by maxjiang23326** [comment]
I'm having the same issue with v5.1.1 -- if the `initialValues` object changes, all the input fields will be overwritten by the new `initialValues` object. 

Will this issue be addressed in v6.0.0?

**Comment by erikras** [comment]
If the past is anything to go by, there will soon be complaining that they were depending on the value overwriting. 😃 I do, however, agree with you guys that it should not overwrite the current form values when `initialValues` changes (if the form as already been initialized). People that want to do that can dispatch `INITIALIZE` manually.

Fix published as `v5.1.2`.

**Comment by kpdecker** [comment]
Thank you!

**Comment by maxjiang23326** [comment]
Thanks a lot!

**Comment by kromit** [comment]
Using 5.1.3 and `value` is now always undefined. Only `initialValue` has the value. All my forms are broken now. I did not expect this from bugfix version change... :(

My form is displayed along with the list and it is expected that all user values are overwritten when new item from the list is selected. There should be a better communication about such breaking changes. At least the old behavior should be the default not the new.

**Comment by erikras** [comment]
@kromit I apologize for that. 😞

**Comment by erikras** [comment]
Okay, guys. @kromit convinced me that initializing every time `initialValues` changes is the way to go. For those of you that don't want that, you will need to specify `overwriteOnInitialValuesChange: false` to negate this behavior. This has been published as [`v5.2.1`](https://github.com/erikras/redux-form/releases/tag/v5.2.1). Apologies for the vacillation.

**Comment by EddyBorja** [comment]
Is there a way to allow an overwrite only for fields that have a different initial value from the previous initialValues?

Currently, if a user is editing a large form and just one initialValue is changed, it seems the entire form re-renders and the user loses whatever input they entered, all for that one value.

**Comment by kpdecker** [comment]
@EddyBorja I believe this is the behavior with the `overwriteOnInitialValuesChange` flag set to false.

**Comment by EddyBorja** [comment]
@kpdecker I tried it, and it seems like the initialValues are never changed from what they were the first time they are set.

In my scenario, my initialValues actually start off as blank strings, because my container needs to fetch data from the DB to populate the initialValues. Once the data comes in, it sets the initial values for the form based off the data it gets back.

However, by setting `overwriteOnInitialValuesChange` to false, the initial values never change from their blank strings, nor when any other updates are made to the data in the DB. 

This is within a Meteor app.

**Comment by erikras** [comment]
Published `enableReinitialize` flag in [`v6.0.0-rc.1`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-rc.1).
